### ISSUE TITLE
Clean project directory structure for ease of manteinance

### ISSUE THREAD

**Issue by aliteral** [DarkPlacesEngine/darkplaces#148]
Hi! Sorry to make an issue for this, but I wanted to ask if it would be possible to define a better project directory structure. It could make working on the project easier on the long run. I made a basic schema:

project directory
 -misc files
 - README
 - .gitignore
 - license
  -- src/core (.c files)
  -- src/include (.h files)
  -- assets (for pngs and related stuff)
 - more misc files
 
 I'm sorry to present it this way, i'm still re-learning some github concepts along the way.

**Comment by Baker7** [comment]
DarkPlaces to me has a nice and simple "1-folder to rule them all" structure that makes working with the engine source easy.

The opposite is fteqw with a huge folder tree with stuff in every folder.  What folder is something in with fteqw source?  Who knows!  You need to use a tool to locate the file.

With DarkPlaces, you know where any file is at all times.

The many folders point of view tends to be a "newer developer thing" that is in love with folder structure, but what really happens with such projects is that the files get moved from folder to folder over time and it is royal mess.

If you want to see some nasty folderitis, look at any Android project where there is a huge rats nest of a folder tree and there are numerous folders that contain 1 or 2 files or a couple of pngs, it requires maybe 45 seconds to locate a file in a file manager.

**Comment by bones-was-here** [comment]
I thought about changing this, but really the only reason is it would look nicer in github's GUI (less scrolling to get to the README).  

I'd do it by moving every .c and .h file to a single subdirectory, maybe moving assets (icons) to a different subdir, and leaving everything else where it is.  That way it will look nicer on github and still be simple.  

However I'd prefer to ask @ladyhavoc first who isn't in chat so pinging her here....

**Comment by aliteral** [comment]
My concern when asking this was modifying the makefile to account for new directories. In fact, I forked the repo to play a little bit with the changes but didn't find a way make the appropiate ones work when compiling...
I will keep digging into to see if I can make it work.

**Comment by aliteral** [comment]
> Folder structure does not reveal where something is, it increases the amount of time required to find files by obscuring their location.

You are correct in this sense, but I think it will be valid to point out that it isn't easy for begginers to understand the project if there are lots of files in only one directory. Also, it makes it easier to find the core .c files and the header files. Now, that is not to say that we should separate everything inside those subdirs. 
What I mean is giving a little bit of structure for ease of manteinance, looking at the fact that the project folder has every file in them, making it look more complex than it should be.

Of course, if it is not important for those that heavily contribute, then I think it is best to keep it the way it is.

**Comment by aliteral** [comment]
Made a fork to tinker with the files and structure, and I'm sorry to ask, but could someone point me in the right direction?
Read and re-read the makefiles and can't figure out where or how to set up the /src/core (%.c) and /src/include (%.h) directories so they can be properly accounted for.
